helix-core/src/main/java/org/apache/helix/util/HelixUtil.java

```diff
@@ -533,21 +533,38 @@ public static long getSystemPropertyAsLong(String propertyKey, long propertyDefa
 
   /**
    * Compose the config for an instance
-   * @param instanceName
+   * @param instanceName the unique name of the instance
    * @return InstanceConfig
    */
+  @Deprecated
   public static InstanceConfig composeInstanceConfig(String instanceName) {
```

Comment:
We can keep this as it use just one string to construct very simple instance config.

Not every company requires complicated instance config.